_studio/hevce_hw/h265/src/mfx_h265_encode_hw_utils.cpp

```diff
@@ -1321,7 +1321,7 @@ void MfxVideoParam::SyncMfxToHeadersParam(mfxU32 numSlicesForSTRPSOpt)
 
 
     slo.max_dec_pic_buffering_minus1    = mfx.NumRefFrame;
-    slo.max_num_reorder_pics            = GetNumReorderFrames(mfx.GopRefDist - 1, isBPyramid(),isField());
+    slo.max_num_reorder_pics            = std::min(GetNumReorderFrames(mfx.GopRefDist - 1, isBPyramid(),isField()), slo.max_dec_pic_buffering_minus1);
```

Comment:
NumReorderFrames more than DPB size means that encoder parameters check/configuration logic was broken and correct encoding is impossible already.
Clipping values in packed headers will not help here.

Reply:
OK, the change can be reverted: it doesn't affect the new functionality.